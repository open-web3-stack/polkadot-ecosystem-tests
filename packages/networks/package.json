{
	"name": "@e2e-test/networks",
	"type": "module",
	"dependencies": {
		"@acala-network/chopsticks-testing": "^0.13.1",
		"@e2e-test/shared": "workspace:*"
	},
	"main": "./dist/cjs/index.js",
	"module": "./dist/esm/index.js",
	"files": [
		"*",
		"!tsconfig.json",
		"!**/*.test.ts"
	],
	"exports": {
		".": {
			"require": "./dist/cjs/index.js",
			"import": "./dist/esm/index.js",
			"default": "./dist/esm/index.js"
		},
		"./chains": {
			"require": "./dist/cjs/chains/index.js",
			"import": "./dist/esm/chains/index.js",
			"default": "./dist/esm/chains/index.js"
		},
		"./*": {
			"require": "./dist/cjs/*.js",
			"import": "./dist/esm/*.js",
			"default": "./dist/esm/*.js"
		}
	}
}
