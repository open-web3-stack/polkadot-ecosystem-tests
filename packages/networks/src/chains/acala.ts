import { defaultAccount } from '@e2e-test/shared'

import { defineChain } from '../defineChain.js'

const custom = {
  acala: {
    relayToken: 'DOT',
    relayLiquidToken: 'LDOT',
    stableToken: 'AUSD',

    paraAccount: '13YMK2eYoAvStnzReuxBjMrAvPXmmdsURwZvc62PrdXimbNy',
    dot: { Token: 'DOT' },
    ldot: { Token: 'LDOT' },
    dai: { Erc20: '0x54a37a01cd75b616d63e0ab665bffdb0143c52ae' },
    wbtc: { ForeignAsset: 5 },
    ausd: { Token: 'AUSD' },
    aca: { Token: 'ACA' },
    lcdot: { LiquidCrowdloan: 13 },
  },
  karura: {
    relayToken: 'KSM',
    relayLiquidToken: 'LKSM',
    stableToken: 'KUSD',

    paraAccount: '13YMK2eYoAvStnzReuxBjMrAvPXmmdsURwZvc62PrdXimbNy',
    ksm: { Token: 'KSM' },
    lksm: { Token: 'LKSM' },
    usdt: { ForeignAsset: 7 },
    rmrk: { ForeignAsset: 0 },
    dai: { Erc20: '0x4bb6afb5fa2b07a5d1c499e1c3ddb5a15e709a71' },
    ausd: { Token: 'KUSD' },
    kar: { Token: 'KAR' },
  },
}

const getInitStorages = (config: typeof custom.acala | typeof custom.karura) => ({
  System: {
    account: [[[defaultAccount.alice.address], { providers: 4, data: { free: 10 * 1e12 } }]],
  },
  Tokens: {
    accounts: [
      [[defaultAccount.alice.address, { Token: config.relayToken }], { free: 10 * 1e12 }],
      [[defaultAccount.alice.address, { Token: config.relayLiquidToken }], { free: 100 * 1e12 }],
      [[defaultAccount.alice.address, { Token: config.stableToken }], { free: 1000 * 1e12 }],
    ],
  },
  Sudo: {
    key: defaultAccount.alice.address,
  },
  EvmAccounts: {
    accounts: [[['0x82a258cb20e2adb4788153cd5eb5839615ece9a0'], defaultAccount.alice.address]],
    evmAddresses: [[[defaultAccount.alice.address], '0x82a258cb20e2adb4788153cd5eb5839615ece9a0']],
  },
  Homa: {
    // avoid impact test outcome
    $removePrefix: ['redeemRequests', 'unbondings', 'toBondPool'],
    // so that bump era won't trigger unbond
    relayChainCurrentEra: 100,
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers', 'supportedVersion'],
  },
})

export const acala = defineChain({
  name: 'acala',
  endpoint: 'wss://acala-rpc.aca-api.network',
  paraId: 2000,
  custom: custom.acala,
  initStorages: getInitStorages(custom.acala),
})

export const karura = defineChain({
  name: 'karura',
  endpoint: 'wss://karura-rpc.aca-api.network',
  paraId: 2000,
  custom: custom.karura,
  initStorages: getInitStorages(custom.karura),
})
