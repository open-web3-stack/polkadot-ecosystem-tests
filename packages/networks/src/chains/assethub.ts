import { defaultAccounts } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  assetHubPolkadot: {
    dot: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
  assetHubKusama: {
    ksm: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
}

const getInitStorages = (config: typeof custom.assetHubPolkadot | typeof custom.assetHubKusama) => ({
  System: {
    account: [[[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }]],
  },
  Assets: {
    account: [
      [[config.usdtIndex, defaultAccounts.alice.address], { balance: 1000e6 }], // USDT
    ],
  },
})

export const assetHubPolkadot = defineChain({
  name: 'assetHubPolkadot',
  endpoint: 'wss://rpc-asset-hub-polkadot.luckyfriday.io',
  paraId: 1000,
  custom: custom.assetHubPolkadot,
  initStorages: getInitStorages(custom.assetHubPolkadot),
})

export const assetHubKusama = defineChain({
  name: 'assetHubKusama',
  endpoint: 'wss://rpc-asset-hub-kusama.luckyfriday.io',
  paraId: 1000,
  custom: custom.assetHubKusama,
  initStorages: getInitStorages(custom.assetHubKusama),
})
