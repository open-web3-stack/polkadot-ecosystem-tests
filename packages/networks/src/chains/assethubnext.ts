import { defaultAccounts, defaultAccountsSr25519 } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  assetHubNextWestend: {
    wnd: { Concrete: { parents: 1, interior: 'Here' } },
    usdt: { Concrete: { parents: 0, interior: { X2: [{ PalletInstance: 50 }, { GeneralIndex: 1984 }] } } },
    usdtIndex: 1984,
  },
}

const getInitStorages = (config: typeof custom.assetHubNextWestend) => ({
  System: {
    account: [
      [[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }],
      [[defaultAccountsSr25519.alice.address], { providers: 1, data: { free: 1000e10 } }],
    ],
  },
  Assets: {
    account: [
      [[config.usdtIndex, defaultAccounts.alice.address], { balance: 1000e6 }], // USDT
    ],
  },
})

export const assetHubNextWestend = defineChain({
  name: 'assetHubNextWestend',
  endpoint: 'wss://westend-asset-hub-next-rpc.parity-chains-scw.parity.io',
  paraId: 1100,
  custom: custom.assetHubNextWestend,
  initStorages: getInitStorages(custom.assetHubNextWestend),
})
