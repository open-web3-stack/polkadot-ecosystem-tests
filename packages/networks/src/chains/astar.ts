import { defaultAccounts } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  astar: {
    relayToken: '340282366920938463463374607431768211455',
    aUSDToken: '18446744073709551617',

    paraAccount: '13YMK2eZzuFY1WZGagpYtTgbWBWGdoUD2CtrPj1mQPjY8Ldc',
    dot: 340282366920938463463374607431768211455n,
    astr: { Concrete: { parents: 0, interior: 'Here' } },
    aca: 18446744073709551616n,
    usdt: 4294969280n,
  },
  shiden: {
    relayToken: '340282366920938463463374607431768211455',
    aUSDToken: '18446744073709551616',

    paraAccount: 'F7fq1jNy74AqkJ1DP4KqSrWtnTGtXfNVoDwFhTvvPxUvJaq',
    ksm: 340282366920938463463374607431768211455n,
    sdn: { Concrete: { parents: 0, interior: 'Here' } },
    kar: 18446744073709551618n,
    usdt: 4294969280n,
  },
}

const getInitStorages = (config: typeof custom.astar | typeof custom.shiden) => ({
  System: {
    account: [[[defaultAccounts.alice.address], { providers: 1, data: { free: '100000000000000000000' } }]],
  },
  Assets: {
    account: [
      [[config.relayToken, defaultAccounts.alice.address], { balance: 10 * 1e12 }],
      [[config.aUSDToken, defaultAccounts.alice.address], { balance: 10 * 1e12 }],
    ],
  },
  Sudo: {
    key: defaultAccounts.alice.address,
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers', 'supportedVersion'],
  },
})

export const astar = defineChain({
  name: 'astar',
  paraId: 2006,
  endpoint: 'wss://astar-rpc.dwellir.com',
  custom: custom.astar,
  initStorages: getInitStorages(custom.astar),
})

export const shiden = defineChain({
  name: 'shiden',
  paraId: 2007,
  endpoint: 'wss://shiden-rpc.dwellir.com',
  custom: custom.shiden,
  initStorages: getInitStorages(custom.shiden),
})
