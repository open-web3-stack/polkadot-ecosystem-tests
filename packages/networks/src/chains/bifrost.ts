import { defineChain } from '../defineChain.js'
import { defaultAccounts, defaultAccountsSr25519, testAccounts } from '../testAccounts.js'

const custom = {
  bifrostPolkadot: {
    relayToken: 'DOT',
    bnc: { Native: 'BNC' },
  },
  bifrostKusama: {
    relayToken: 'KSM',
    bnc: { Token: 'BNC' },
  },
}

const getInitStorages = (_config: typeof custom.bifrostPolkadot | typeof custom.bifrostKusama) => ({
  System: {
    account: [
      [[defaultAccounts.alice.address], { providers: 1, data: { free: 666e12 } }],
      [[defaultAccountsSr25519.alice.address], { providers: 1, data: { free: 666e12 } }],
      [[testAccounts.alice.address], { providers: 1, data: { free: 666e12 } }],
    ],
  },
  PolkadotXcm: {
    // avoid sending xcm version change notifications to makes things faster
    $removePrefix: ['versionNotifyTargets', 'versionNotifiers'],
  },
})

export const bifrostPolkadot = defineChain({
  name: 'bifrostPolkadot',
  endpoint: 'wss://bifrost-polkadot.ibp.network',
  paraId: 2030,
  custom: custom.bifrostPolkadot,
  initStorages: getInitStorages(custom.bifrostPolkadot),
})

export const bifrostKusama = defineChain({
  name: 'bifrostKusama',
  endpoint: 'wss://us.bifrost-rpc.liebi.com/ws',
  paraId: 2001,
  custom: custom.bifrostKusama,
  initStorages: getInitStorages(custom.bifrostKusama),
})
