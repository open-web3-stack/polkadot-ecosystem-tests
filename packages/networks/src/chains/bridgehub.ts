import { defaultAccounts } from '../defaultAccounts.js'
import { defineChain } from '../defineChain.js'

const custom = {
  bridgeHubPolkadot: {
    dot: { Concrete: { parents: 1, interior: 'Here' } },
  },
  bridgeHubKusama: {
    ksm: { Concrete: { parents: 1, interior: 'Here' } },
  },
}

const getInitStorages = (_config: typeof custom.bridgeHubPolkadot | typeof custom.bridgeHubKusama) => ({
  System: {
    account: [[[defaultAccounts.alice.address], { providers: 1, data: { free: 1000e10 } }]],
  },
})

export const bridgeHubPolkadot = defineChain({
  name: 'bridgeHubPolkadot',
  endpoint: 'wss://polkadot-bridge-hub-rpc.polkadot.io',
  paraId: 1002,
  custom: custom.bridgeHubPolkadot,
  initStorages: getInitStorages(custom.bridgeHubPolkadot),
})

export const bridgeHubKusama = defineChain({
  name: 'bridgeHubKusama',
  endpoint: 'wss://kusama-bridge-hub-rpc.polkadot.io',
  paraId: 1002,
  custom: custom.bridgeHubKusama,
  initStorages: getInitStorages(custom.bridgeHubKusama),
})
