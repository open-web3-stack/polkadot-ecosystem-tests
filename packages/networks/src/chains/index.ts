export * from './acala.js'
export * from './assethub.js'
export * from './astar.js'
export * from './hydraDX.js'
export * from './moonbeam.js'
export * from './polkadot.js'
