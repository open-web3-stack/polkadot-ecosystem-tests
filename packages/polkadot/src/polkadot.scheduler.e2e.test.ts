import { polkadot } from '@e2e-test/networks/chains'
import { baseSchedulerE2ETests, registerTestTree } from '@e2e-test/shared'

registerTestTree(baseSchedulerE2ETests(polkadot, { testSuiteName: 'Polkadot Scheduler' }))
