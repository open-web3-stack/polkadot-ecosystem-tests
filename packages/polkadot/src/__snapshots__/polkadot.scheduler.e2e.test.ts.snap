// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Polkadot Scheduler > cancel named scheduled task with wrong origin > events when cancelling named task with insufficient origin 1`] = `
[
  {
    "data": {
      "dispatchError": "BadOrigin",
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 50000)",
          "refTime": "(rounded 800000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Polkadot Scheduler > cancel scheduled task with wrong origin > events when cancelling task with insufficient origin 1`] = `
[
  {
    "data": {
      "dispatchError": "BadOrigin",
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 50000)",
          "refTime": "(rounded 800000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Polkadot Scheduler > cancelling a named scheduled task is possible > events for scheduled task cancellation 1`] = `
[
  {
    "data": {
      "index": 0,
      "when": "(redacted)",
    },
    "method": "Canceled",
    "section": "scheduler",
  },
  {
    "data": {
      "id": null,
      "result": "Ok",
      "task": "(redacted)",
    },
    "method": "Dispatched",
    "section": "scheduler",
  },
]
`;

exports[`Polkadot Scheduler > cancelling a scheduled task is possible > events for scheduled task cancellation 1`] = `
[
  {
    "data": {
      "index": 0,
      "when": "(redacted)",
    },
    "method": "Canceled",
    "section": "scheduler",
  },
  {
    "data": {
      "id": null,
      "result": "Ok",
      "task": "(redacted)",
    },
    "method": "Dispatched",
    "section": "scheduler",
  },
]
`;

exports[`Polkadot Scheduler > execution of scheduled preimage lookup call works > events for scheduled lookup-task execution 1`] = `
[
  {
    "data": {
      "new_": "(redacted)",
      "old": "(redacted)",
    },
    "method": "TotalIssuanceForced",
    "section": "balances",
  },
  {
    "data": {
      "id": null,
      "result": "Ok",
      "task": "(redacted)",
    },
    "method": "Dispatched",
    "section": "scheduler",
  },
]
`;

exports[`Polkadot Scheduler > execution of scheduled preimage lookup call works > note preimage events 1`] = `
[
  {
    "data": {
      "hash_": "(hash)",
    },
    "method": "Noted",
    "section": "preimage",
  },
]
`;

exports[`Polkadot Scheduler > priority-based execution of weighted tasks works correctly > events for priority weighted tasks execution 1`] = `
[
  {
    "data": {
      "new_": "(redacted)",
      "old": "(redacted)",
    },
    "method": "TotalIssuanceForced",
    "section": "balances",
  },
  {
    "data": {
      "id": null,
      "result": "Ok",
      "task": "(redacted)",
    },
    "method": "Dispatched",
    "section": "scheduler",
  },
  {
    "data": {
      "when": "(redacted)",
    },
    "method": "AgendaIncomplete",
    "section": "scheduler",
  },
]
`;

exports[`Polkadot Scheduler > schedule named task with wrong origin > events when scheduling named task with insufficient origin 1`] = `
[
  {
    "data": {
      "dispatchError": "BadOrigin",
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 50000)",
          "refTime": "(rounded 700000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Polkadot Scheduler > schedule task with wrong origin > events when scheduling task with insufficient origin 1`] = `
[
  {
    "data": {
      "dispatchError": "BadOrigin",
      "dispatchInfo": {
        "class": "Normal",
        "paysFee": "Yes",
        "weight": {
          "proofSize": "(rounded 50000)",
          "refTime": "(rounded 600000000)",
        },
      },
    },
    "method": "ExtrinsicFailed",
    "section": "system",
  },
]
`;

exports[`Polkadot Scheduler > scheduling a call is possible, and the call itself succeeds > events for scheduled task execution 1`] = `
[
  {
    "data": {
      "new_": "(redacted)",
      "old": "(redacted)",
    },
    "method": "TotalIssuanceForced",
    "section": "balances",
  },
  {
    "data": {
      "id": null,
      "result": "Ok",
      "task": "(redacted)",
    },
    "method": "Dispatched",
    "section": "scheduler",
  },
]
`;

exports[`Polkadot Scheduler > scheduling a call using a preimage that gets removed > events for failing scheduled lookup-task execution 1`] = `
[
  {
    "data": {
      "id": null,
      "task": "(redacted)",
    },
    "method": "CallUnavailable",
    "section": "scheduler",
  },
]
`;

exports[`Polkadot Scheduler > scheduling a call using a preimage that gets removed > note preimage events 1`] = `
[
  {
    "data": {
      "hash_": "(hash)",
    },
    "method": "Noted",
    "section": "preimage",
  },
]
`;

exports[`Polkadot Scheduler > scheduling a named call is possible, and the call itself succeeds > events for scheduled task execution 1`] = `
[
  {
    "data": {
      "new_": "(redacted)",
      "old": "(redacted)",
    },
    "method": "TotalIssuanceForced",
    "section": "balances",
  },
  {
    "data": {
      "id": "(hash)",
      "result": "Ok",
      "task": "(redacted)",
    },
    "method": "Dispatched",
    "section": "scheduler",
  },
]
`;

exports[`Polkadot Scheduler > scheduling a named periodic task that executes multiple times > events for named periodic task execution 1 1`] = `
[
  {
    "data": {
      "new_": "(redacted)",
      "old": "(redacted)",
    },
    "method": "TotalIssuanceForced",
    "section": "balances",
  },
  {
    "data": {
      "id": "(hash)",
      "result": "Ok",
      "task": "(redacted)",
    },
    "method": "Dispatched",
    "section": "scheduler",
  },
  {
    "data": {
      "index": 0,
      "when": "(redacted)",
    },
    "method": "Scheduled",
    "section": "scheduler",
  },
]
`;

exports[`Polkadot Scheduler > scheduling a named periodic task that executes multiple times > events for named periodic task execution 2 1`] = `
[
  {
    "data": {
      "new_": "(redacted)",
      "old": "(redacted)",
    },
    "method": "TotalIssuanceForced",
    "section": "balances",
  },
  {
    "data": {
      "id": "(hash)",
      "result": "Ok",
      "task": "(redacted)",
    },
    "method": "Dispatched",
    "section": "scheduler",
  },
  {
    "data": {
      "index": 0,
      "when": "(redacted)",
    },
    "method": "Scheduled",
    "section": "scheduler",
  },
]
`;

exports[`Polkadot Scheduler > scheduling a named periodic task that executes multiple times > events for named periodic task execution 3 1`] = `
[
  {
    "data": {
      "new_": "(redacted)",
      "old": "(redacted)",
    },
    "method": "TotalIssuanceForced",
    "section": "balances",
  },
  {
    "data": {
      "id": "(hash)",
      "result": "Ok",
      "task": "(redacted)",
    },
    "method": "Dispatched",
    "section": "scheduler",
  },
]
`;

exports[`Polkadot Scheduler > scheduling a named task after a delay is possible > events for scheduled task execution 1`] = `
[
  {
    "data": {
      "new_": "(redacted)",
      "old": "(redacted)",
    },
    "method": "TotalIssuanceForced",
    "section": "balances",
  },
  {
    "data": {
      "id": "(hash)",
      "result": "Ok",
      "task": "(redacted)",
    },
    "method": "Dispatched",
    "section": "scheduler",
  },
]
`;

exports[`Polkadot Scheduler > scheduling a named task after a delay is possible > events when scheduling task with delay 1`] = `
[
  {
    "data": {
      "index": 0,
      "when": "(redacted)",
    },
    "method": "Scheduled",
    "section": "scheduler",
  },
  {
    "data": {
      "id": null,
      "result": "Ok",
      "task": "(redacted)",
    },
    "method": "Dispatched",
    "section": "scheduler",
  },
]
`;

exports[`Polkadot Scheduler > scheduling a periodic task is possible > events for  periodic task execution 1 1`] = `
[
  {
    "data": {
      "new_": "(redacted)",
      "old": "(redacted)",
    },
    "method": "TotalIssuanceForced",
    "section": "balances",
  },
  {
    "data": {
      "id": null,
      "result": "Ok",
      "task": "(redacted)",
    },
    "method": "Dispatched",
    "section": "scheduler",
  },
  {
    "data": {
      "index": 0,
      "when": "(redacted)",
    },
    "method": "Scheduled",
    "section": "scheduler",
  },
]
`;

exports[`Polkadot Scheduler > scheduling a periodic task is possible > events for  periodic task execution 2 1`] = `
[
  {
    "data": {
      "new_": "(redacted)",
      "old": "(redacted)",
    },
    "method": "TotalIssuanceForced",
    "section": "balances",
  },
  {
    "data": {
      "id": null,
      "result": "Ok",
      "task": "(redacted)",
    },
    "method": "Dispatched",
    "section": "scheduler",
  },
  {
    "data": {
      "index": 0,
      "when": "(redacted)",
    },
    "method": "Scheduled",
    "section": "scheduler",
  },
]
`;

exports[`Polkadot Scheduler > scheduling a periodic task is possible > events for  periodic task execution 3 1`] = `
[
  {
    "data": {
      "new_": "(redacted)",
      "old": "(redacted)",
    },
    "method": "TotalIssuanceForced",
    "section": "balances",
  },
  {
    "data": {
      "id": null,
      "result": "Ok",
      "task": "(redacted)",
    },
    "method": "Dispatched",
    "section": "scheduler",
  },
]
`;

exports[`Polkadot Scheduler > scheduling a task after a delay is possible > events for scheduled task execution 1`] = `
[
  {
    "data": {
      "new_": "(redacted)",
      "old": "(redacted)",
    },
    "method": "TotalIssuanceForced",
    "section": "balances",
  },
  {
    "data": {
      "id": null,
      "result": "Ok",
      "task": "(redacted)",
    },
    "method": "Dispatched",
    "section": "scheduler",
  },
]
`;

exports[`Polkadot Scheduler > scheduling a task after a delay is possible > events when scheduling task with delay 1`] = `
[
  {
    "data": {
      "index": 0,
      "when": "(redacted)",
    },
    "method": "Scheduled",
    "section": "scheduler",
  },
  {
    "data": {
      "id": null,
      "result": "Ok",
      "task": "(redacted)",
    },
    "method": "Dispatched",
    "section": "scheduler",
  },
]
`;

exports[`Polkadot Scheduler > scheduling an overweight call is possible, but the call itself fails > events when scheduling overweight task 1`] = `
[
  {
    "data": {
      "id": null,
      "task": "(redacted)",
    },
    "method": "PermanentlyOverweight",
    "section": "scheduler",
  },
]
`;

exports[`Polkadot Scheduler > setting and canceling retry configuration for named scheduled tasks > events for failed named task execution 1`] = `
[
  {
    "data": {
      "id": "(hash)",
      "result": {
        "Err": "BadOrigin",
      },
      "task": "(redacted)",
    },
    "method": "Dispatched",
    "section": "scheduler",
  },
  {
    "data": {
      "index": 0,
      "when": "(redacted)",
    },
    "method": "Scheduled",
    "section": "scheduler",
  },
]
`;

exports[`Polkadot Scheduler > setting and canceling retry configuration for named scheduled tasks > events for retry config cancellation 1`] = `
[
  {
    "data": {
      "id": null,
      "result": {
        "Err": {
          "Module": {
            "error": "0x01000000",
            "index": 1,
          },
        },
      },
      "task": "(redacted)",
    },
    "method": "Dispatched",
    "section": "scheduler",
  },
]
`;

exports[`Polkadot Scheduler > setting and canceling retry configuration for unnamed scheduled tasks > events for failed task execution 1`] = `
[
  {
    "data": {
      "id": null,
      "result": {
        "Err": "BadOrigin",
      },
      "task": "(redacted)",
    },
    "method": "Dispatched",
    "section": "scheduler",
  },
  {
    "data": {
      "index": 0,
      "when": "(redacted)",
    },
    "method": "Scheduled",
    "section": "scheduler",
  },
]
`;

exports[`Polkadot Scheduler > setting and canceling retry configuration for unnamed scheduled tasks > events for retry config cancellation 1`] = `
[
  {
    "data": {
      "id": null,
      "task": "(redacted)",
    },
    "method": "RetryCancelled",
    "section": "scheduler",
  },
  {
    "data": {
      "id": null,
      "result": "Ok",
      "task": "(redacted)",
    },
    "method": "Dispatched",
    "section": "scheduler",
  },
]
`;
